tests/integration/data_sources_and_expectations/expectations/test_expect_query_results_to_match_source.py

```diff
@@ -312,3 +455,186 @@ def test_expect_query_results_to_match_source_error(multi_source_batch: MultiSou
+            value=RenderedAtomicValue(
+                schema={"type": "TableType"},
+                header_row=[
+                    RendererTableValue(
+                        schema=RendererSchema(type=RendererValueType.STRING),
+                        value=col_name,
+                    )
+                    for col_name in ["e", "f", "g", "h"]
+                ],
+                table=[
+                    [
+                        RendererTableValue(
+                            schema=RendererSchema(type=RendererValueType.STRING),
+                            value=str(value),
+                        )
+                        for _ in ["e", "f", "g", "h"]
+                    ]
+                    for value in [4, 5, 6]
+                ],
+            ),
+            value_type="TableType",
+        ),
+        RenderedAtomicContent(
+            name=AtomicDiagnosticRendererType.OBSERVED_VALUE,
+            value=RenderedAtomicValue(
+                schema={"type": "TableType"},
+                header_row=[
+                    RendererTableValue(
+                        schema=RendererSchema(type=RendererValueType.STRING),
+                        value=col_name,
+                    )
+                    for col_name in ["a", "b", "c", "d"]
+                ],
+                table=[
+                    [
+                        RendererTableValue(
+                            schema=RendererSchema(type=RendererValueType.STRING),
+                            value=str(value),
+                        )
+                        for _ in ["a", "b", "c", "d"]
+                    ]
+                    for value in [1, 2, 3]
+                ],
+            ),
+            value_type="TableType",
+        ),
+    ]
+
+
+@multi_source_batch_setup(
+    multi_source_test_configs=SQLITE_ONLY,
+    source_data=pd.DataFrame({"foo": [1, 2, 3, 3]}),
+    target_data=pd.DataFrame({"bar": [1, 4, 5, 5]}),
+)
+def test_rendering_with_one_column(multi_source_batch: MultiSourceBatch):
+    source_table = multi_source_batch.source_table_name
+    result = multi_source_batch.target_batch.validate(
+        gxe.ExpectQueryResultsToMatchSource(
+            source_data_source_name=multi_source_batch.source_data_source_name,
+            source_query=f"SELECT foo FROM {source_table}",
+            target_query="SELECT bar FROM {batch}",
+        )
+    )
+    result.render()
+
+    assert result.rendered_content == [
+        RenderedAtomicContent(
+            name=AtomicDiagnosticRendererType.OBSERVED_VALUE,
+            value_type="StringValueType",
+            value=RenderedAtomicValue(
+                schema={"type": "com.superconductive.rendered.string"},
+                meta_notes={"format": MetaNotesFormat.STRING, "content": []},
+                template="$ov__0 $ov__1 $ov__2 $exp__0 $exp__1 $exp__2",
+                params={
+                    "expected_value": {
```

Comment:
are `expected_value` and `observed_value` special keys? If not, I'll probably rename to `missing_values` and `unexpected_values`

Reply:
No they are not. That should be fine.